### PR TITLE
Properly render vehicle rental layer with 0 stations

### PR DESCRIPTION
**Pull request by evansiroky**
Also fix the vehicle rental storybook stories and improve the storybook example to show detailed controlling of the vehicle rental layer via handlers of layer control events.

### CODE REVIEW

**Review by miles-grant-ibigroup — APPROVED** [review]

File: packages/vehicle-rental-overlay/src/leaflet-layer-control-interface.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+   */
+  visible: boolean;
+};
+
+/**
+ * This class is a necessary intermediary to handle events from the Leaflet map
+ * existing in the parent element. See more notes on the other properties and
+ * methods for how everything interacts.
+ */
+export default class LeafletLayerControlInterface extends Component<
```
Comment (binh-dam-ibigroup) [inline packages/vehicle-rental-overlay/src/leaflet-layer-control-interface.tsx:34]: (non-blocking) There seem to be two opposite approaches, one taken in OTP-UI by e.g. `ParkAndRideOverlay`, `StopsOverlay`, and `TransitVehicleOverlay` that doesn't keep the `visible` state of the layer, and this one, also used in OTP-RR by `ConnectedVehicleRentalOverlay` and `GtfsRtVehicleOverlay`... I am still missing something about the two approaches, but long term I would favor an approach similar to `ParkAndRideOverlay`.

**Review by binh-dam-ibigroup — APPROVED** [review]

**Comment by evansiroky** [comment]
> (non-blocking) There seem to be two opposite approaches, one taken in OTP-UI by e.g. `ParkAndRideOverlay`, `StopsOverlay`, and `TransitVehicleOverlay` that doesn't keep the `visible` state of the layer, and this one, also used in OTP-RR by `ConnectedVehicleRentalOverlay` and `GtfsRtVehicleOverlay`... I am still missing something about the two approaches, but long term I would favor an approach similar to `ParkAndRideOverlay`.

After looking at the code, I completely agree with what @binh-dam-ibigroup is saying here. The OTP-RR approach of handling visibility via state should not be used at all. Therefore, I'm closing this PR and superseding it with https://github.com/opentripplanner/otp-ui/pull/269.

**Comment by evansiroky** [comment]
> > (non-blocking) There seem to be two opposite approaches, one taken in OTP-UI by e.g. `ParkAndRideOverlay`, `StopsOverlay`, and `TransitVehicleOverlay` that doesn't keep the `visible` state of the layer, and this one, also used in OTP-RR by `ConnectedVehicleRentalOverlay` and `GtfsRtVehicleOverlay`... I am still missing something about the two approaches, but long term I would favor an approach similar to `ParkAndRideOverlay`.
> 
> After looking at the code, I completely agree with what @binh-dam-ibigroup is saying here. The OTP-RR approach of handling visibility via state should not be used at all. Therefore, I'm closing this PR and superseding it with #269.

All right, hopefully for a final time, I'm changing my mind and reopening this again as I now see the purpose of handling visibility via state in the `LeafletLayerControlInterface` component. I have refactored the comments of the `LeafletLayerControlInterface` class in 585c0f690db70b5dfa3c68be04ce2522749f1b82.

File: packages/vehicle-rental-overlay/src/VehicleRentalOverlay.story.js [hunk 2]
```diff
@@ -34,7 +36,7 @@ const MyCircle = ({ fillColor = "gray", pixels, strokeColor }) => {
       {children}
     </CircleMarker>
   );
-  GeneratedCircle.coreUtils.types.propTypes = {
+  GeneratedCircle.propTypes = {
```
Comment (evansiroky) [inline packages/vehicle-rental-overlay/src/VehicleRentalOverlay.story.js:39]: @landonreed, this is thing that was causing the problems that you were asking about in https://github.com/opentripplanner/otp-ui/pull/268#issuecomment-883335088

**Review by landonreed — APPROVED** [review]
LGTM. Thanks for your diligence on this one, @evansiroky. I'm pretty happy with this approach. My only question is do we need to make this interface change to the other overlays as well (and if so, how do we want to share this common code)?

**Comment by evansiroky** [comment]
> LGTM. Thanks for your diligence on this one, @evansiroky. I'm pretty happy with this approach. My only question is do we need to make this interface change to the other overlays as well (and if so, how do we want to share this common code)?

We could... but that's probably not the next most largest value add thing we could do.

**Pull request merged**